packages/lib/src/application/App.ts

```diff
@@ -0,0 +1,158 @@
+    const instance: ContractWrapper = await this._copyContract(packageName, contractName, contractClass);
+    await this._initNonUpgradeableInstance(instance, contractClass, packageName, contractName, initMethodName, initArgs);
+    return instance;
+  }
+
+  public async createProxy(contractClass: ContractFactory, packageName: string, contractName: string, initMethodName: string, initArgs?: string[]): Promise<ContractWrapper> {
+    const { receipt } = typeof(initArgs) === 'undefined'
+      ? await this._createProxy(packageName, contractName)
+      : await this._createProxyAndCall(contractClass, packageName, contractName, initMethodName, initArgs);
+
+    log.info(`TX receipt received: ${receipt.transactionHash}`);
+    const address: string = _.findLast(receipt.logs, (l) => l.event === 'ProxyCreated').args.proxy;
+    log.info(`${packageName} ${contractName} proxy: ${address}`);
+    return contractClass.at(address);
+  }
+
+  public async upgradeProxy(proxyAddress: string, contractClass: ContractFactory, packageName: string, contractName: string, initMethodName: string, initArgs: any): Promise<ContractWrapper> {
+    const { receipt }: any = typeof(initArgs) === 'undefined'
+      ? await this._upgradeProxy(proxyAddress, packageName, contractName)
+      : await this._upgradeProxyAndCall(proxyAddress, contractClass, packageName, contractName, initMethodName, initArgs);
+    log.info(`TX receipt received: ${receipt.transactionHash}`);
+    return contractClass.at(proxyAddress);
+  }
+
+  private async _createProxy(packageName: string, contractName: string): Promise<any> {
+    log.info(`Creating ${packageName} ${contractName} proxy without initializing...`);
+    const initializeData: string = '';
+    return sendTransaction(this.appContract.create, [packageName, contractName, initializeData], this.txParams);
+  }
+
+  private async _createProxyAndCall(contractClass: ContractFactory, packageName: string, contractName: string, initMethodName: string, initArgs: any): Promise<any> {
+    const { method: initMethod, callData }: CalldataInfo = buildCallData(contractClass, initMethodName, initArgs);
+    log.info(`Creating ${packageName} ${contractName} proxy and calling ${callDescription(initMethod, initArgs)}`);
+    return sendTransaction(this.appContract.create, [packageName, contractName, callData], this.txParams);
+  }
+
+  private async _upgradeProxy(proxyAddress: string, packageName: string, contractName: string): Promise<any> {
+    log.info(`Upgrading ${packageName} ${contractName} proxy without running migrations...`);
+    return sendTransaction(this.appContract.upgrade, [proxyAddress, packageName, contractName], this.txParams);
+  }
+
+  private async _upgradeProxyAndCall(proxyAddress: string, contractClass: ContractFactory, packageName: string, contractName: string, initMethodName: string, initArgs: any): Promise<any> {
```

Comment:
We don't know what type sendTransaction returns, but considering that we know the Solidity code of the contract being called in these funcitons,  we could specify the return type of sendTransaction calls in them. However, it would be time consuming to do now and it'd really just be a nice to have at this point. If you want to do it, great, otherwise, take this as something to consider for later.